src/tdb_encode_model.c

```diff
@@ -411,11 +412,13 @@ tdb_error make_grams(FILE *grouped,
     TDB_TIMER_END("encode_model/find_candidates")
 
     /* collect frequencies of *all* occurring bigrams of candidate unigrams */
-    TDB_TIMER_START
-    ret = event_fold(all_bigrams, grouped, num_events, items, num_fields, &g);
-    if (ret)
-        goto done;
-    TDB_TIMER_END("encode_model/all_bigrams")
+    if (!no_bigrams) {
```

Comment:
The `choose_grams` function below is an `O(n^2)` operation, which should be the most expensive part here. If there are no bigrams, the `choose_grams` could be made much less expensive.